src/engine_redraw.c

```diff
@@ -99,21 +99,22 @@ static void draw_creature_view_icons(struct Thing* creatng)
         ps_units_per_px = (22 * units_per_pixel) / spr->SHeight;
         y = MyScreenHeight - scale_ui_value_lofi(spr->SHeight * 2);
     }
-    struct CreatureControl* cctrl = creature_control_get_from_thing(creatng);
-    for (SpellKind Spell = SplK_Freeze; Spell <= SplK_TimeBomb; Spell++)
-    {
-        if (creature_affected_by_spell(creatng, Spell))
+    struct CreatureControl *cctrl = creature_control_get_from_thing(creatng);
+    struct SpellConfig *spconf;
+    for (SpellKind spell_idx = 0; spell_idx < game.conf.magic_conf.spell_types_count; spell_idx++)
+    { // ^Maybe it should only loop with the actives spells? -> 'for (SpellKind spell_idx = 0; spell_idx < CREATURE_MAX_SPELLS_CASTED_AT; spell_idx++)'
```

Comment:
That would not work. That would check the first 5 spells, and will not check if spells 6~30 are active,

Reply:
I assumed it was looping through 'spkind', not casted_spells[spell_idx]

Reply:
yes, looks fine.